token-service/src/main/java/org/p2p/token/service/repository/metadata/TokenMetadataRemoteRepository.kt

```diff
@@ -25,29 +25,31 @@ internal class TokenMetadataRemoteRepository(
     private val tokenServiceUrl
         get() = urlProvider.loadTokenServiceEnvironment().baseServiceUrl
 
-    override suspend fun loadTokensMetadata(lastModified: String?): UpdateTokenMetadataResult =
-        withContext(dispatchers.io) {
-            val response = try {
-                api.getZipFile("${tokenServiceUrl}get_all_tokens_info", lastModified)
-            } catch (e: HttpException) {
-                return@withContext if (e.code() == HttpURLConnection.HTTP_NOT_MODIFIED) {
-                    UpdateTokenMetadataResult.NoUpdate
-                } else {
-                    UpdateTokenMetadataResult.Error(e)
-                }
+    override suspend fun loadTokensMetadata(
+        ifModifiedSince: String?
+    ): UpdateTokenMetadataResult = withContext(dispatchers.io) {
+        val response = try {
+            api.getZipFile("${tokenServiceUrl}get_all_tokens_info", ifModifiedSince)
+        } catch (e: HttpException) {
+            return@withContext if (e.code() == HttpURLConnection.HTTP_NOT_MODIFIED) {
```

Comment:
i think return@withContext is not really needed here

Reply:
why? 
We should return the function. 

The code below expects the response only